app/services/burn-event/service.go

```diff
@@ -0,0 +1,178 @@
+)
+
+type Service struct {
+	thresholdAccount hedera.AccountID
+	payerAccount     hedera.AccountID
+	repository       repository.BurnEvent
+	hederaNodeClient client.HederaNode
+	mirrorNodeClient client.MirrorNode
+	logger           *log.Entry
+}
+
+func NewService(
+	thresholdAccount string,
+	payerAccount string,
+	hederaNodeClient client.HederaNode,
+	mirrorNodeClient client.MirrorNode,
+	repository repository.BurnEvent) *Service {
+
+	threshold, err := hedera.AccountIDFromString(thresholdAccount)
+	if err != nil {
+		log.Fatalf("Invalid bridge threshold account: [%s].", thresholdAccount)
+	}
+
+	payer, err := hedera.AccountIDFromString(payerAccount)
+	if err != nil {
+		log.Fatalf("Invalid payer account: [%s].", payerAccount)
+	}
+
+	return &Service{
+		thresholdAccount: threshold,
+		payerAccount:     payer,
+		repository:       repository,
+		hederaNodeClient: hederaNodeClient,
+		mirrorNodeClient: mirrorNodeClient,
+		logger:           config.GetLoggerFor("Burn Event Service"),
+	}
+}
+
+func (s Service) ProcessEvent(event burn_event.BurnEvent) {
+	err := s.repository.Create(event.TxHash, event.Amount, event.Recipient.String())
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to create a burn event record. Error [%s].", event.TxHash, err)
+		return
+	}
+
+	var transactionResponse *hedera.TransactionResponse
+	if event.NativeToken == "HBAR" {
+		transactionResponse, err = s.hederaNodeClient.
+			SubmitScheduledHbarTransferTransaction(event.Amount, event.Recipient, s.thresholdAccount, s.payerAccount, event.TxHash)
+	} else {
+		tokenID, err := hedera.TokenIDFromString(event.NativeToken)
+		if err != nil {
+			s.logger.Errorf("[%s] - failed to parse native token [%s] to TokenID. Error [%s].", event.TxHash, event.NativeToken, err)
+			return
+		}
+		transactionResponse, err = s.hederaNodeClient.
+			SubmitScheduledTokenTransferTransaction(event.Amount, tokenID, event.Recipient, s.thresholdAccount, s.payerAccount, event.TxHash)
+	}
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to submit scheduled transaction. Error [%s].", event.TxHash, err)
+		return
+	}
+
+	s.logger.Infof("[%s] - Successfully submitted scheduled transaction [%s] for [%s] to receive [%d] tinybars.",
+		event.TxHash,
+		transactionResponse.TransactionID, event.Recipient, event.Amount)
+
+	txReceipt, err := transactionResponse.GetReceipt(s.hederaNodeClient.GetClient())
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to get transaction receipt for [%s]", event.TxHash, transactionResponse.TransactionID)
+		return
+	}
+
+	switch txReceipt.Status {
+	case hedera.StatusIdenticalScheduleAlreadyCreated:
+		s.handleScheduleSign(event.TxHash, helper.ToMirrorNodeTransactionID(txReceipt.ScheduledTransactionID.String()), *txReceipt.ScheduleID)
+	case hedera.StatusSuccess:
+		transactionID := helper.ToMirrorNodeTransactionID(txReceipt.ScheduledTransactionID.String())
+		s.logger.Infof("[%s] - Updating db status to Submitted with TransactionID [%s].", event.TxHash, transactionID)
```

Comment:
I think that it is better to log the update after it has occurred. That way we will keep it consistent as-well